src/Sarif/CodeGenHints.json

```diff
@@ -34,6 +34,30 @@
       }
     }
   ],
+  "AnnotatedCodeLocation.Essential": [
+    {
+      "kind": "AttributeHint",
+      "arguments": {
+        "typeName": "JsonProperty",
+        "namespaceName": "Newtonsoft.Json",
+        "arguments": [
+          "\"essential\""
+        ],
+        "properties": {
+          "DefaultValueHandling": "DefaultValueHandling.Ignore"
```

Comment:
This says "don't write it out if it has the default value". The problem is that if it _doesn't_ have the default value, it _will_ be written out. So a result with essential==true will be written out the same way, which we don't want. It might be better to just put a `JsonIgnore` attribute here (never read it in, never write it out). That would present a challenge for our "corrective optics". I'm tempted to leave this as is until we get to the corrective optics phase.